api/http_client.go

```diff
@@ -0,0 +1,161 @@
+package api
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"time"
+)
+
+// originally from https://github.com/leonelquinteros/hubspot/blob/master/client.go
+
+// ClientConfig object used for client creation
+type ClientConfig struct {
+	APIHost     string
+	APIUsername string
+	APIPassword string
+	HTTPTimeout time.Duration
+	DialTimeout time.Duration
+	TLSTimeout  time.Duration
+}
+
+// NewClientConfig constructs a ClientConfig object with the environment variables set as default
+func NewClientConfig() ClientConfig {
+	apiHost := "http://localhost:3000"
+	var apiUsername string
+	var apiPassword string
+
+	if os.Getenv("HOST") != "" {
```

Comment:
There is no real reason to check if the os.Getenv returns an empty string, as not setting the variable will end up having an empty string too.

```go
var target string // target = ""

if os.Getenv("HOST") != "" { // os.Getenv("HOST") = ""
  target = os.Getenv("HOST")
}

// target = "" still

```